--- oai-marc	2022-03-28 11:14:26.877390116 +0200
+++ oai-marc.594	2022-04-05 18:34:20.889018863 +0200
@@ -1261,33 +1261,31 @@
 		#			write_error(header.identifier(), '245', SIF, '189', 'Chybí popis vedlejší role v 245c.')
 
 		# << >> TAG count
-		R=False
-		for F in metadata.get_fields('506'):
-			if '190' in F.get_subfields('a') or '191' in F.get_subfields('a'): R=True
-		if not R and '700' in metadata:
-			if '245' in metadata and 'c' in metadata['245']:
-				BRACKET = len(re.findall('<<[^<>]*>>', metadata['245']['c']))
-				for F in metadata.get_fields('505'):
-					if 'r' in F: BRACKET += len(re.findall('<<[^<>]*>>', F['r']))
-				if BRACKET > 0:
-					FC = len(metadata.get_fields('100','593'))
-					for F in metadata.get_fields('700'):
-						if metadata.leader[7] == 'm':
-							if len(F.get_subfields('4')) == 1 and F['4'] in ['aqt','ctb']: continue
-						FC+=1
-					if FC != BRACKET:
-						write_error(header.identifier(), '245', SIF, '190', 'Počet tagů v podpoli 245c neodpovídá počtu polí.')
+		if '700' in metadata and '245' in metadata and 'c' in metadata['245']:
+			BRACKET = len(re.findall('<<[^<>]*>>', metadata['245']['c']))
+			for F in metadata.get_fields('505'):
+				if 'r' in F: BRACKET += len(re.findall('<<[^<>]*>>', F['r']))
+			for F in metadata.get_fields('594'):
+				if 'x' in F: BRACKET += 1
+			if BRACKET > 0:
+				FC = 0
+				for F in metadata.get_fields('100','700'):
+					if len(F.get_subfields('4')) == 1 and F['4'] in ['aqt','ctb']: continue
+					FC+=1
+				if FC != BRACKET:
+					write_error(header.identifier(), '245', SIF, '190', 'Počet tagů v podpoli 245c neodpovídá počtu polí.')
 
 		# TAG value
-		if not R and '245' in metadata and 'c' in metadata['245']:
+		if '245' in metadata and 'c' in metadata['245']:
 			for T in re.findall('<<[^<>]*>>', metadata['245']['c']):
 				if '100' in metadata and 'a' in metadata['100']:
 					if T == '<<' + metadata['100']['a'] + '>>': continue
 				for F in metadata.get_fields('700'):
 					if 'a' in F and T == '<<' + F['a'] + '>>': break
 				else:
-					if not (T == '<<>>' or '[=' in T or T[2].isupper()):
-						write_error(header.identifier(), '245', SIF, '191', 'Neplatný tag v podpoli 245c.')
+					if T != '<<>>' and T[2].islower():
+						 if not re.match('.*' + T + '[^<>]+\[=.*', metadata['245']['c']):
+							write_error(header.identifier(), '245', SIF, '191', 'Neplatný tag v podpoli 245c.')
 
 		# CPK
 		if int(re.sub('^.*-(\d+)$', '\\1', header.identifier())) >= 2735900:
@@ -1301,6 +1299,22 @@
 			if 'b' in F and not re.match('^\d{9}$', F['b']):
 				write_error(header.identifier(), '994', SIF, '193', "Podpole LKR-b obsahuje neplatné systémové číslo.")
 
+		# 954
+		for F in metadata.get_fields('594'):
+			for F2 in metadata.get_fields('100','700'):
+				if F2.tag == '700' and len(F2.get_subfields('4')) == 1 and F2['4'] in ['aqt','ctb']: continue
+				if len(F.subfields) >= 2:
+					if F.subfields[0] == 'x' and F.subfields[2:] == F2.subfields: break
+			else:
+				write_error(header.identifier(), '594', SIF, '194', "Pole 954 se neshoduje.")
+			if not metadata.get_fields('100','110','111','700','710','711'):
+				for F3 in metadata.get_fields('500'):
+					if 'a' in F3 and F3['a'] == 'Nepodepsáno.': break
+				else:
+					write_error(header.identifier(), '594', SIF, '195', "Neplatné pole 954.")
+			if 'a' and F and 'y' in F:
+				if F['a'] != F['y']:
+					write_error(header.identifier(), '594', SIF, '196', "Podpole 954a a 954y se neshoduje.")
+
 	COUNTER+=1
 
 # EXIT -------------------
